virttest/qemu_devices/qdevices.py

```diff
@@ -1498,12 +1499,12 @@ def verify_hotplug(self, out, monitor):
 class Dimm(QDevice):
     """
     PC-Dimm or NVDIMM device, support for memory hotplug using the device and
-    the QOM objects 'memory-backend-ram' and 'memory-backend-file'
+    QOM objects.
     """
 
-    __attributes__ = ["memdev", "slot", "addr", "node", "unarmed"]
-
     def __init__(self, params=None, dimm_type='pc-dimm'):
+        attrs = get_dev_attrs('device', dimm_type, params)
```

Comment:
Could we use `self.attrs` ?  Then it is an attribute of an instance, no need to get it again when it is needed, such as the change with `MemoryHotplugTest` in this patch.